Libraries/LibMedia/PlaybackManager.cpp

```diff
@@ -71,33 +70,80 @@ DecoderErrorOr<NonnullRefPtr<PlaybackManager>> PlaybackManager::try_create(Reado
     if (supported_video_tracks.is_empty() && supported_audio_tracks.is_empty())
         return DecoderError::with_description(DecoderErrorCategory::NotImplemented, "No supported video or audio tracks found"sv);
 
-    RefPtr<AudioMixingSink> audio_sink = nullptr;
-    if (!supported_audio_tracks.is_empty())
-        audio_sink = DECODER_TRY_ALLOC(AudioMixingSink::try_create());
+    auto preferred_video_track = demuxer->get_preferred_track_for_type(TrackType::Video).value_or({});
+    if (preferred_video_track.has_value() && !supported_video_tracks.contains_slow(*preferred_video_track))
+        preferred_video_track = {};
+    auto preferred_audio_track = demuxer->get_preferred_track_for_type(TrackType::Audio).value_or({});
+    if (preferred_audio_track.has_value() && !supported_audio_tracks.contains_slow(*preferred_audio_track))
+        preferred_audio_track = {};
 
-    // Create the time provider.
-    auto time_provider = DECODER_TRY_ALLOC([&] -> ErrorOr<NonnullRefPtr<MediaTimeProvider>> {
-        if (audio_sink)
-            return TRY(try_make_ref_counted<WrapperTimeProvider<AudioMixingSink>>(*audio_sink));
-        return TRY(try_make_ref_counted<GenericTimeProvider>());
-    }());
+    auto duration = demuxer->total_duration().value_or(AK::Duration::zero());
+
+    main_thread_event_loop.deferred_invoke([weak_playback_manager, on_complete = move(on_complete), video_tracks = move(supported_video_tracks), video_track_datas = move(supported_video_track_datas), preferred_video_track, audio_tracks = move(supported_audio_tracks), audio_track_datas = move(supported_audio_track_datas), preferred_audio_track, duration] mutable {
+        auto playback_manager = weak_playback_manager->take_strong();
+        if (!playback_manager)
+            return;
+
+        RefPtr<AudioMixingSink> audio_sink = nullptr;
+        if (!audio_tracks.is_empty())
+            audio_sink = AudioMixingSink::try_create().release_value_but_fixme_should_propagate_errors();
```

Comment:
Feel free to MUST() this for now, and we can probably do away with the `try_` part later.